File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        freq = {}
        for i in range(n):
            if nums[i] in freq:
                freq[nums[i]] += 1
            else:
                freq[nums[i]] = 1

        bucket = [[] for j in range(n+1)]

        for num, f in freq.items():
            bucket[f].append(num)
        res = []
        for t in range(n, 0, -1):
            for num in bucket[t]:
                res.append(num)
                if len(res) == k:
                    return res

File: Data Structures & Algorithms/find-all-numbers-disappeared-in-an-array/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    findDisappearedNumbers(nums) {
        const set = new Set(nums);
        const res = []
        for (let i = 1; i < (nums.length + 1); i++) {
            if (!set.has(i)) {
                res.push(i)
            }
        }
        return res;
    }
}
